src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -48,10 +71,11 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
                 this.course = this.courseCalculationService.getCourse(this.courseId);
             });
         }
-        this.groupExercises(this.DUE_DATE_DESC);
 
-        this.translateSubscription = this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
-            this.groupExercises(this.DUE_DATE_DESC);
+        this.applyFiltersAndOrder();
```

Comment:
What happens if open Artemis in two windows and change the order in the other window.
This would update the localStorage in an inconsistent matter, wouldn't it?
Maybe, in a new service handling this interaction  (see above) we should [handle the browser events fired for changes in local storage?](https://stackoverflow.com/a/57227771/6324877)

Reply:
Well, If one window would have checked "Hide overdue" and the other would uncheck it, then it just would be unchecked on a reload of the page. In the end, the most recent change is persisted, so from a user's perspective, I think it still makes sense. But you are definitely right regarding refactoring the local storage into its own service in a future PR